core/src/main/java/org/elasticsearch/action/get/MultiGetResponse.java

```diff
@@ -149,6 +166,65 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    public static MultiGetResponse fromXContent(XContentParser parser) throws IOException {
+        String currentFieldName = null;
+        List<MultiGetItemResponse> items = new ArrayList<>();
+        for (Token token = parser.nextToken(); token != Token.END_OBJECT; token = parser.nextToken()) {
+            switch (token) {
+                case FIELD_NAME:
+                    currentFieldName = parser.currentName();
+                    break;
+                case START_ARRAY:
+                    if (Fields.DOCS.equals(currentFieldName)) {
+                        for (token = parser.nextToken(); token != Token.END_ARRAY; token = parser.nextToken()) {
+                            if (token == Token.START_OBJECT) {
+                                String index = null;
+                                String type = null;
+                                String id = null;
+                                ElasticsearchException exception = null;
+                                GetResult getResult = null;
+                                item: for (token = parser.nextToken(); token != Token.END_OBJECT; token = parser.nextToken()) {
+                                    switch (token) {
```

Comment:
Same here, maybe have a default case? Not sure if it needs to do anything (I think parsing should be lenient for the response parsing)

Reply:
No, there is no default case. Normally I would add a default case that throws an exception, but we should be lenient here.